lib/compat/wordpress-5.9/get-global-styles-and-settings.php

```diff
@@ -102,26 +102,44 @@ function wp_get_global_stylesheet( $types = array() ) {
 			}
 		}
 
-		$supports_theme_json = WP_Theme_JSON_Resolver_Gutenberg::theme_has_support();
-		$supports_link_color = get_theme_support( 'experimental-link-color' );
+		$tree = WP_Theme_JSON_Resolver::get_merged_data();
+
+		$supports_theme_json = WP_Theme_JSON_Resolver::theme_has_support();
 		if ( empty( $types ) && ! $supports_theme_json ) {
 			$types = array( 'variables', 'presets' );
 		} elseif ( empty( $types ) ) {
 			$types = array( 'variables', 'styles', 'presets' );
 		}
 
-		$origins = array( 'default', 'theme', 'custom' );
-		if ( ! $supports_theme_json && ! $supports_link_color ) {
-			// In this case we only enqueue the core presets (CSS Custom Properties + the classes).
-			$origins = array( 'default' );
-		} elseif ( ! $supports_theme_json && $supports_link_color ) {
-			// For the legacy link color feature to work, the CSS Custom Properties
-			// should be in scope (either the core or the theme ones).
-			$origins = array( 'default', 'theme' );
+		/*
+		 * If variables are part of the stylesheet,
+		 * we add them for all origins (default, theme, user).
+		 * This is so themes without a theme.json still work as before 5.9:
+		 * they can override the default presets.
+		 * See https://core.trac.wordpress.org/ticket/54782
+		 */
+		$styles_variables = '';
+		if ( in_array( 'variables', $types, true ) ) {
```

Comment:
Changeset matches the core patch except this line: https://github.com/WordPress/wordpress-develop/pull/2233/files#diff-1be8a8089d87d1b4b6e449c0967680d6f7c170d13f247180a975cbea5e54070fR121.

Was there a specific reason for setting the `strict` param in `in_array` here?

Reply:
I just check the trunk though and it seems you have added in another PR? I'm trying to find it 😄

Reply:
Changes were [done](https://github.com/WordPress/wordpress-develop/commit/c6c3d9b928b08408918c8651f3125ec15974289b) after that PR and the current code from core uses the strict check.

Reply:
Got it: https://github.com/WordPress/wordpress-develop/commit/c6c3d9b928b08408918c8651f3125ec15974289b